index.bs

```diff
@@ -130,6 +130,71 @@ much like a data property as possible. Specific guidance in this regard includes
 
+<h3 id="live-vs-static">Consider whether objects should be live or static</h3>
+
+Objects returned from functions, attribute getters, etc.,
+can either be live or static.
+A <dfn>live object</dfn> is one that continues to reflect
+changes made after it was returned to the caller.
+A <dfn>static object</dfn> is one that reflects
+the state at the time it was returned.
+
+Objects that are the way state is mutated are generally live.
+For example, DOM nodes are returned as live objects,
+since they are the API through which attributes are set and other changes
+to the DOM are made.
+They also reflect changes to the DOM made in other ways
+(such as through user interaction with forms).
+Even though the nodes themselves are live,
+they might be returned from a static object,
+such as a static {{NodeList}} that is
+the result of {{ParentNode/querySelectorAll()}}.
+
+Objects that represent a collection that might change over time
+should generally be returned as static objects.
+This is because it is confusing to users of the API when
+a collection changes while being iterated.
+Because of this,
+it is generally considered a mistake that methods like
+{{Document/getElementsByTagName()}} return live objects;
+{{ParentNode/querySelectorAll()}} was made to return static objects
+as a result of this experience.
+
+Note:  It's possible that some of this advice should be reconsidered
+for <a>Iterable</a>, <a>Maplike</a>, and <a>Setlike</a> types
+which can be iterated using iterators that have
+well-defined behavior for mutation that happens during iteration.
+(Is this true for all three?)
+This point likely needs further discussion,
+and perhaps further experience of use of these types.
+
+It's also worth considering the implications of having
+live versus static objects for the speed of implementations of the API.
+When the data needed by an object are expensive to compute up-front,
+there is an advantage for that object to be live so that the results
+can be computed lazily, such as for {{Window/getComputedStyle()}}.
+On the other hand,
+if the data needed by an object are expensive to keep up-to-date,
+such as for {{ParentNode/querySelectorAll()}},
+then providing a static object avoids
+having to keep the object updated until it is garbage collected
+(which may be substantially after its last use).
+
+Likewise, the choice of live versus static objects
+can influence the memory use of an API.
+If each call of a method returns a new static object,
+and the objects are large,
+then substantial amounts of memory can be wasted
+until the next garbage collection.
+
+The choice of whether an object is live or static may also
+influence whether it should be returned
+from an attribute getter or from a method.
+See [[#attributes-like-data]].
+In particular,
+if a result that changes frequently is returned as a static object,
+it should probably be returned from a method rather than an attribute getter.
+
```

Comment:
Maybe not as part of this PR or even this section, but a common point of confusion is the case of "read only lists" (where only the implementation can change them). Currently we tend to recommend people return static snapshots as arrays (for methods, when they change frequently) or frozen arrays (for attributes, when they change infrequently). It's not the best solution, compared to some kind of real read-only array type, but we tend to be going in that direction. Notably we are moving away from creating various fake-array types like NodeList or StyleSheetList for this purpose.